srcs/execute/env_utils.c

```diff
@@ -0,0 +1,102 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   env_utils.c                                        :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: jsaariko <jsaariko@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2020/10/28 11:19:34 by jsaariko      #+#    #+#                 */
+/*   Updated: 2020/10/29 13:51:00 by jsaariko      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "execute.h"
+#include "error.h"
+
+t_env		*get_env_item(char *env_str)
+{
+	t_env	*item;
+	int		i;
+
+	item = (t_env *)e_malloc(sizeof(t_env));
+	i = 0;
+	while (env_str[i] != '\0' && env_str[i] != '=')
+		i++;
+	item->key = (char *)e_malloc(sizeof(char) * (i + 1));
+	ft_strlcpy(item->key, env_str, i + 1);
+	if (env_str[i] != '=')
+		item->value = NULL;
+	else
+	{
+		item->value = ft_strdup(env_str + i + 1);
+		if (item->value == NULL)
+			error_exit_errno();
+	}
+	return (item);
+}
+
+t_vector	*envp_to_env(char **envp)
+{
+	size_t		i;
+	size_t		count;
+	t_env		*cur;
+	t_vector	*env;
+	int			ret;
+
+	i = 0;
+	count = 0;
+	cur = NULL;
+	while (envp[count])
+		count++;
+	env = (t_vector *)e_malloc(sizeof(t_vector));
+	vector_init(env);
+	while (i < count)
+	{
+		cur = get_env_item(envp[i]);
+		ret = vector_push(env, cur);
+		if (ret == 0)
+			error_exit_errno();
+		i++;
+	}
+	return (env);
+}
+
+static char	*create_envp_str(t_env *cur)
+{
+	char	*final;
+	char	*tmp;
+	char	*tmp2;
+
+	tmp = ft_strdup(cur->key);
+	if (tmp == NULL)
+		error_exit_errno();
+	tmp2 = ft_strjoin(tmp, "=");
+	if (tmp2 == NULL)
+		error_exit_errno();
+	final = ft_strjoin(tmp2, cur->value);
+	if (final == NULL)
+		error_exit_errno();
+	free(tmp);
+	tmp = NULL;
+	free(tmp2);
+	tmp = NULL;
+	return (final);
+}
+
+char		**env_to_envp(t_vector *env)
+{
```

Comment:
This function is never called in the current set-up. Is that intentional?

Reply:
Also please never interpret these questions as passive aggressive hahaha... I was genuinely wondering whether it was included for the sake of being a useful utility we might need later, or whether an existing function was supposed to call it...

Reply:
We will need them in the future as running our shell will require for us to swap between differently formatted environments frequently. If it winds up being an issue, I can always delete and rewrite